compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java

```diff
@@ -574,6 +574,9 @@ void addFunctionType(DiagnosticPos pos, Set<Whitespace> ws, boolean paramsAvail,
         if (paramsAvail) {
             functionTypeNode.addWS(commaWsStack.pop());
             functionTypeNode.params.addAll(this.varListStack.pop());
+        } else if (restParamAvail) {
+            // VarListStack pops when function typed param has only rest params
```

Comment:
Do we need this comment? Shall we maybe improve it to indicate popping out the empty list added to the var list stack if no non-rest params are available?

Reply:
Thanks. Resolved